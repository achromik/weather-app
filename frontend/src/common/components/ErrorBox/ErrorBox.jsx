import React from 'react';
import { Alert } from 'antd';

export const ErrorBox = ({ message }) => (
    <div>
        <Alert banner showIcon type="error" message={message} closable />
    </div>
);
