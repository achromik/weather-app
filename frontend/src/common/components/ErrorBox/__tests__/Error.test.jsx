import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { ErrorBox } from '../ErrorBox';

const mockMessage = 'City not found';

describe('ErrorBox component', () => {
    it('renders', () => {
        const wrapper = shallow(<ErrorBox message={mockMessage} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders passed message', () => {
        const wrapper = shallow(<ErrorBox message={mockMessage} />);
        expect(wrapper.find('Alert').props().message).toBe('City not found');
    });

    it('renders as closable alert', () => {
        const wrapper = shallow(<ErrorBox message={mockMessage} />);
        expect(wrapper.find('Alert').props().closable).toBe(true);
    });
});
