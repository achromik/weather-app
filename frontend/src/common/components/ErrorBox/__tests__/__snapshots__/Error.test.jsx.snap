// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBox component renders 1`] = `
<div>
  <Alert
    banner={true}
    closable={true}
    message="City not found"
    showIcon={true}
    type="error"
  />
</div>
`;
